core/tests/test_utils.py

```diff
@@ -2048,6 +2052,68 @@ def utcnow(cls):
         finally:
             setattr(datetime, 'datetime', original_datetime_type)
 
+    @contextlib.contextmanager
```

Comment:
I don't think the new things you've added here should go in the general test class. We should perhaps have a separate base test class for audit jobs?

Reply:
Yes, updated.